pkg/controller/job/job_controller.go

```diff
@@ -123,11 +124,13 @@ type Controller struct {
 	recorder    record.EventRecorder
 
 	podUpdateBatchPeriod time.Duration
+
+	clock clock.WithTicker
 }
 
 // NewController creates a new Job controller that keeps the relevant pods
 // in sync with their corresponding Job objects.
-func NewController(podInformer coreinformers.PodInformer, jobInformer batchinformers.JobInformer, kubeClient clientset.Interface) *Controller {
+func NewController(podInformer coreinformers.PodInformer, jobInformer batchinformers.JobInformer, kubeClient clientset.Interface, clock clock.WithTicker) *Controller {
```

Comment:
For the purpose of changing this clock only in unit tests I'd NOT expose this here, but only inside the `Controller` struct, to which unit tests have access since that's how we modify other bits, like podController or alike.

Reply:
not sure if it can be done without explicitly recreating the queues. Is that fine?

Reply:
Since there is this open question and this is not urgent, we can leave it for after code freeze.

@harshanarayana can you make sure put this change through after code freeze?

Reply:
@alculquicondor Sure. I can do that. Do you want me to create a different Issue to track this or just use the current one itself for all of it? Since we have no conclusion on this yet, I am open to either of the options. Or hold this changes from getting in until we conclude and address the required changes that @soltysh has in mind.

Reply:
just the current one. Let's wait for Maciej's answer.

Reply:
I remember solving a similar issue by introduction of the interim function `NewPodGCInternal` used to call from tests with non-default values for `gcCheckPeriod` and `quarantineTime`: https://github.com/kubernetes/kubernetes/blob/96fd7b1b50362858a621c03a16e21b71f64cfcbf/pkg/controller/podgc/gc_controller.go#L82